realm/src/main/java/io/realm/android/RealmCursor.java

```diff
@@ -0,0 +1,757 @@
+    private static final int FIELD_TYPE_NULL;
+    private static final int FIELD_TYPE_INTEGER;
+    private static final int FIELD_TYPE_FLOAT;
+    private static final int FIELD_TYPE_STRING;
+    private static final int FIELD_TYPE_BLOB;
+    static {
+        if (android.os.Build.VERSION.SDK_INT >= 11) {
+            FIELD_TYPE_NULL = Cursor.FIELD_TYPE_NULL;
+            FIELD_TYPE_INTEGER = Cursor.FIELD_TYPE_INTEGER;
+            FIELD_TYPE_FLOAT = Cursor.FIELD_TYPE_FLOAT;
+            FIELD_TYPE_STRING = Cursor.FIELD_TYPE_STRING;
+            FIELD_TYPE_BLOB = Cursor.FIELD_TYPE_BLOB;
+        } else {
+            FIELD_TYPE_NULL = 0;
+            FIELD_TYPE_INTEGER = 1;
+            FIELD_TYPE_FLOAT = 2;
+            FIELD_TYPE_STRING = 3;
+            FIELD_TYPE_BLOB = 4;
+        }
+    }
+
+    private final Realm realm;
+    private TableOrView table;
+    private int rowIndex;
+    private boolean closed;
+    private long idColumnIndex = -1; // Column index for the "_id" field. -1 means it hasn't been set
+    private final DataSetObservable dataSetObservable = new DataSetObservable();
+    private RealmChangeListener changeListener = new RealmChangeListener() {
+        @Override
+        public void onChange() {
+            if (dataSetObservable != null) {
+                dataSetObservable.notifyChanged(); // TODO Should be replaced with something more fine grained.
+            }
+        }
+    };
+
+
+    /**
+     * Exposes a query result as a cursor. Use {@link RealmResults#getCursor()} instead of this
+     * constructor
+     *
+     * @param table Table view representing the query results.
+     */
+    public RealmCursor(Realm realm, TableOrView table) {
+        this.table = table;
+        this.rowIndex = -1;
+        this.realm = realm;
+        realm.addChangeListener(changeListener);
+    }
+
+    /**
+     * Returns the numbers of rows in the cursor.
+     *
+     * @return the number of rows in the cursor.
+     */
+    @Override
+    public int getCount() {
+        checkClosed();
+        return (int) table.size();
+    }
+
+    /**
+     * Returns the current position of the cursor in the row set.
+     * The value is zero-based. When the row set is first returned the cursor
+     * will be at position -1, which is before the first row. After the
+     * last row is returned another call to next() will leave the cursor past
+     * the last entry, at a position of count().
+     *
+     * @return the current cursor position.
+     */
+    @Override
+    public int getPosition() {
+        checkClosed();
+        return rowIndex;
+    }
+
+    /**
+     * Move the cursor by a relative amount, forward or backward, from the
+     * current position. Positive offsets move forwards, negative offsets move
+     * backwards. If the final position is outside of the bounds of the result
+     * set then the resultant position will be pinned to -1 or count() depending
```

Comment:
"resultant" -> "new" or "final" (the word "resultant" isn't wrong but sounds a bit old-fashioned to me).